arkouda/numeric.py

```diff
@@ -2001,3 +2027,76 @@ def value_counts(
     (array([0, 2, 4]), array([3, 2, 1]))
     """
     return GroupBy(pda).count()
+
+
+@typechecked
+def clip(
+    pda: pdarray,
+    lo: Union[numeric_scalars, pdarray],
+    hi: Union[numeric_scalars, pdarray],
+) -> pdarray:
+    """
+    Mimic the behavior of numpy.clip.
```

Comment:
Can you make the doc string more similar to this:
https://github.com/numpy/numpy/blob/v1.26.0/numpy/core/fromnumeric.py#L2100-L2169

And this: 
https://bears-r-us.github.io/arkouda/_modules/arkouda/numeric.html#value_counts

Reply:
This style guide might help: https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
This is a lot better, but still doesn't render exactly right.  Maybe try something like this:
```   
    Clip (limit) the values in an array to a given range [lo,hi].

    Given an array a, values outside the range are clipped to the
    range edges, such that all elements lie in the range.

    There is no check on lo < hi, and if lo > hi, the corresponding
    value of the array will be set to hi.

    Parameters
    ----------
    pda : pdarray
        The values to clip.
    lo  : scalar or pdarray
        The lower value(s) to clip to.
    hi  : scalar or pdarray
        The higher value(s) to clip to.

    Notes
    -----
    Either lo or hi may be None, but not both.
    If lo > hi, all x = hi.

    If all inputs are int, output is int, but if any input is real, output is real.

    Returns
    -------
    arkouda.pdarrayclass.pdarray
        A pdarray matching pda, except that element x remains x if lo <= x <= hi,
                                                or becomes lo if x < lo,
                                                or becomes hi if x > hi.

        If lo or hi (or both) are pdarrays, the check is by pairwise elements.
        See examples.

    Examples
    --------
    >>> a = ak.array([1,2,3,4,5,6,7,8,9,10])
    >>> ak.clip(a,3,8)
    array([3,3,3,4,5,6,7,8,8,8])
    >>> ak.clip(a,3,8.0)
    array([3.00000000000000000 3.00000000000000000 3.00000000000000000 4.00000000000000000 5.00000000000000000 ...
        6.00000000000000000 7.00000000000000000 8.00000000000000000 8.00000000000000000 8.00000000000000000])
    >>> ak.clip(a,None,7)
    array([1,2,3,4,5,6,7,7,7,7])
    >>> ak.clip(a,5,None)
    array([5,5,5,5,5,6,7,8,9,10])
    >>> ak.clip(a,None,None)
    ValueError : either min or max must be supplied
    >>> ak.clip(a,ak.array([2,2,3,3,8,8,5,5,6,6],8))
    array([2,2,3,4,8,8,7,8,8,8])
    >>> ak.clip(a,4,ak.array([10,9,8,7,6,5,5,5,5,5]))
    array([4,4,4,4,5,5,5,5,5,5])   Note that hi < lo for much of this array ; see Notes

    Raises
    ------
    ValueError
        Raised if both lo and hi are None.
```

Some tips:

- Use capital letters and punctuation when possible.
- The placement of tabs matters a lot, so follow the examples.  For example, in the Parameters section, the first line will be `parameter_name : type`, and the second line will be tab indented description of the parameter.
- The returns section should start with the type returned, and the description should be on the following line, tab indented.
- The Section names should be standard ("Parameters", "Attributes", etc...).  The following line of hyphens needs to be the exact same length as the section name.  So "Examples" will be followed by 8 hyphens because "Examples" has 8 letters.

Reply:
The example section  is still not rendering correctly.  If you replace
```
    Examples:
    --------
```
with
```
    Examples
    --------
```
it should work.  Sphinx is pretty fussy unfortunately.  Other thank that it looks good.

Reply:
Works now.  Thanks!